src/app/related-search/related-search.component.spec.ts

```diff
@@ -79,7 +50,19 @@ describe('RelatedSearchComponent', () => {
     fixture.detectChanges();
   });
 
-  it('should create', () => {
+  it('should create an instance', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should display results if any, along with right keyword', () => {
+    let knowledgeService = fixture.debugElement.injector.get(KnowledgeapiService);
+    let compiled = fixture.debugElement.nativeElement;
+    if (component.results.length > 0) {
```

Comment:
any update regarding this?
A test should have a definite path, we should avoid having condtionals inside tests.
I remembered commenting regarding this at https://github.com/fossasia/susper.com/pull/575#pullrequestreview-47794656 and gave suggestions as below:
just try removing the conditional statement, you will find that tests fail, everytime. 
solution to this is:-
1) directly initialize results variable as shown in above image.
2) have a spy on search service ( https://angular.io/guide/testing#test-a-component-with-an-async-service ) and get results by spying upon that